SWLOR.Game.Server/Feature/BuffTimer.cs

```diff
@@ -0,0 +1,220 @@
+﻿using SWLOR.Game.Server.Core;
+using SWLOR.Game.Server.Core.NWScript.Enum;
+using SWLOR.Game.Server.Service;
+using SWLOR.Game.Server.Service.StatusEffectService;
+using System;
+using System.Collections.Generic;
+
+namespace SWLOR.Game.Server.Feature
+{
+    public static class BuffTimer
+    {
+        /// <summary>
+        /// Get the information for different status effects and report to the player
+        /// when the icon(s) are clicked.
+        /// </summary>
+        [NWNEventHandler("mod_gui_event")]
+        public static void DisplayBuffInfo()
+        {
+            var player = GetLastGuiEventPlayer();
+            var type = GetLastGuiEventType();
+            if (!GetIsPC(player)) return;
+            if (type != GuiEventType.EffectIconClick) return;
+
+            var buffInt = GetLastGuiEventInteger();
+
+            // We don't care about duration for concentration powers, because their duration is arbitrary
+
+            switch(buffInt)
+            {
+                case (int)EffectIconType.Slow:
```

Comment:
This is going to be hard to maintain as we add more effects or change the meaning around the icons. I'd recommend just pulling the name off the 2DA file, should be a one-liner instead.